indico/modules/events/controllers/preparation.py

```diff
@@ -0,0 +1,46 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2020 CERN
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see the
+# LICENSE file for more details.
+
+
+from uuid import uuid4
+
+from flask import jsonify
+from pytz import timezone
+from webargs import fields
+
+from indico.legacy.common.cache import GenericCache
+from indico.util.marshmallow import NaiveDateTime
+from indico.web.args import use_kwargs
+from indico.web.rh import RH
+from indico.web.util import url_for_index
+
+
+class RHPrepareEvent(RH):
+    """Prepare a new event, store it using GenericCache util, and create a UUID."""
+
+    CSRF_ENABLED = False
+
+    @use_kwargs({
+        'title': fields.String(required=True),
+        'start_dt': NaiveDateTime(required=True),
+        'tz': fields.String(),
```

Comment:
`fields.String(required=True, validate=OneOf(common_timezones_set))`

(you can import it from pytz)

Reply:
instead of making it required you can also use `missing=config.DEFAULT_TIMEZONE` to use the server's default timezone if none has been specified... not sure if it's really useful though, maybe it's cleaner to force people to be explicit..

Reply:
Also added `missing=config.DEFAULT_TIMEZONE`. I think it adds some flexibility.